modules/lib/src/main/resources/assets/js/app/ui2/dialog/ConfirmationDialog.tsx

```diff
@@ -50,14 +52,110 @@ const ConfirmationDialogContent = forwardRef<HTMLDivElement, ConfirmationDialogC
     ({defaultSubmitEnabled = true, className, children, ...props}, ref): ReactElement => {
         return (
             <Dialog.Content ref={ref}
-                            className={cn('gap-3 min-w-170 max-w-fit', className)} {...props}>
+                            className={cn('text-main gap-3 min-w-170 max-w-fit', className)} {...props}>
                 <ConfirmationDialogProvider defaultSubmitEnabled={defaultSubmitEnabled}>{children}</ConfirmationDialogProvider>
             </Dialog.Content>
         );
     }
 );
 ConfirmationDialogContent.displayName = 'ConfirmationDialog.Content';
 
+//
+// * ConfirmationDialogGate
+//
+
+const ERROR_DELAY_MS = 500;
+const digitsOnly = (s: string): string => s.replace(/\D+/g, '');
```

Comment:
I don't think this works as intended.
If we expect digits only, that must be enforced at the input level.
Here you're just extracting digits from whatever was entered. But `1_d3` is not `13`, and with this normalization it becomes `13`.
I suggest removing this function entirely.